File: app_tv/src/main/java/tpo/mediaplayer/app_tv/DeviceListAdapter.java
package tpo.mediaplayer.app_tv;

import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class DeviceListAdapter extends RecyclerView.Adapter<DeviceListAdapter.ViewHolder> {
    private Device[] listdata;
    private boolean selected = false;
    private List<Device> toBeDeleted;

    public DeviceListAdapter(Device[] listdata, List<Device> toBeDeleted) {
        this.listdata = listdata;
        this.toBeDeleted = toBeDeleted;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View listItem = layoutInflater.inflate(R.layout.list_item, parent, false);
        listItem.setFocusable(true);
        listItem.setFocusableInTouchMode(true);
        ViewHolder viewHolder = new ViewHolder(listItem);

        return viewHolder;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        final Device device = listdata[position];
        holder.textView.setText(listdata[position].deviceName);
        holder.relativeLayout.setOnClickListener(view -> {
            selected = !selected;
            if (selected) {
                holder.textView.setTextColor(Color.parseColor("#D10000"));
                toBeDeleted.add(device);
            } else {
                holder.textView.setTextColor(Color.parseColor("#FFFFFF"));
                toBeDeleted.remove(device);
            }
            System.out.println(toBeDeleted.toString());
        });
        holder.itemView.setOnFocusChangeListener((v, hasFocus) -> {
            if (hasFocus) {
                holder.textView.setBackgroundColor(Color.parseColor("#FFBB86FC"));
            } else {
                holder.textView.setBackgroundColor(Color.parseColor("#FF3700B3"));
            }
        });
    }

    @Override
    public int getItemCount() {
        return listdata.length;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public TextView textView;
        public RelativeLayout relativeLayout;

        public ViewHolder(View itemView) {
            super(itemView);
            this.textView = (TextView) itemView.findViewById(R.id.textView);
            relativeLayout = itemView.findViewById(R.id.relativeLayout);
        }
    }
}

File: app_tv/src/main/java/tpo/mediaplayer/app_tv/DeviceListAdapter.kt
package tpo.mediaplayer.app_tv

import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import tpo.mediaplayer.app_tv.db.Device

private object DeviceDiffCallback : DiffUtil.ItemCallback<Device>() {
    override fun areItemsTheSame(oldItem: Device, newItem: Device) = oldItem.uid == newItem.uid

    override fun areContentsTheSame(oldItem: Device, newItem: Device) =
        oldItem.uid == newItem.uid
                && oldItem.deviceName == newItem.deviceName
                && oldItem.communicationStr == newItem.communicationStr
}

class NewDeviceListAdapter(private val onClick: (Int) -> Unit) :
    ListAdapter<Device, NewDeviceListAdapter.ViewHolder>(DeviceDiffCallback) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.list_item, parent, false)
        view.isFocusable = true
        view.isFocusableInTouchMode = true
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = getItem(position)!!
        holder.textView.text = item.deviceName
        holder.relativeLayout.setOnClickListener { onClick(item.uid) }
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val textView: TextView = itemView.findViewById(R.id.textView)
        val relativeLayout: RelativeLayout = itemView.findViewById(R.id.relativeLayout)

        init {
            itemView.onFocusChangeListener = View.OnFocusChangeListener { v, hasFocus ->
                if (hasFocus) textView.setBackgroundColor(Color.parseColor("#FFBB86FC"))
                else textView.setBackgroundColor(Color.parseColor("#FF3700B3"))
            }
        }
    }
}